tests/integration/test_poly.py

```diff
@@ -259,6 +261,7 @@ def main() -> None:
     validate(module.compile())
 
 
+@pytest.mark.skip("Not yet supported")
```

Comment:
This worked before, but can no longer compile this since `TypeApply` is gone. We can recover this test case by having smarter type arg inference in the future

Reply:
Exactly!

Reply:
I wonder if there is some syntax we could add in a future PR to let you manually specify those type args and get this example going if type inference doesn't cope and you really, really, want to. But how often does one do this, I don't think it's a priority!

Reply:
I wonder if there is some syntax we could add in a future PR to let you manually specify those type args and get this example going if type inference doesn't cope and you really, really, want to. But how often does one do this, I don't think it's a priority!